src/chainparams.cpp

```diff
@@ -251,13 +254,88 @@ class CTestNetParams : public CChainParams {
     }
 };
 
+/**
+ * Signet
+ */
+class SigNetParams : public CChainParams {
+public:
+    explicit SigNetParams(const ArgsManager& args) {
+        std::vector<uint8_t> bin;
+        vSeeds.clear();
+
+        if (!args.IsArgSet("-signet_blockscript")) {
+            throw std::runtime_error(strprintf("%s: -signet_blockscript is mandatory for signet networks", __func__));
+        }
+        const auto blockscript = args.GetArgs("-signet_blockscript");
+        if (blockscript.size() != 1) {
+            throw std::runtime_error(strprintf("%s: -signet_blockscript cannot be multiple values.", __func__));
+        }
+        bin = ParseHex(blockscript[0]);
+        if (args.IsArgSet("-signet_seednode")) {
+            vSeeds = gArgs.GetArgs("-signet_seednode");
+        }
+
+        LogPrintf("Signet with block script %s\n", blockscript[0]);
```

Comment:
This line causes compilation errors, because of undeclared *blockscript* variable.

```shell
chainparams.cpp: In constructor ‘SigNetParams::SigNetParams(const ArgsManager&)’:
chainparams.cpp:284:52: error: ‘blockscript’ was not declared in this scope
         LogPrintf("Signet with block script %s\n", blockscript[0]);
                                                    ^~~~~~~~~~~
make[2]: *** [Makefile:10152: libbitcoin_common_a-chainparams.o] Error 1
```

Reply:
Thanks, yeah I failed at a rebase. Fixed now.